theSrc/scripts/LabeledScatter.js

```diff
@@ -80,24 +80,26 @@ class LabeledScatter {
     const config = buildConfig(this.data, this.width, this.height)
     try {
       const plot_data = createPlotlyData(config)
-      const plot_layout = createPlotlyLayout(config, this.marginRight(config))
+      const legend_points_and_bubble_legend_width = this.legendPointsAndBubbleLegendWidth(config)
+      const margin_right = legend_points_and_bubble_legend_width > 0 ? legend_points_and_bubble_legend_width : null
+      const plot_layout = createPlotlyLayout(config, margin_right)
       const plot_config = { displayModeBar: false, editable: false }
 
       let plotlyChart = await Plotly.react(this.rootElement, plot_data, plot_layout, plot_config)
       const tmp_layout = {}
-      const is_extra_margin_needed_for_legend = this.isExtraMarginNeededForLegend(plotlyChart._fullLayout, config)
-      if (is_extra_margin_needed_for_legend) {
-        tmp_layout['margin.r'] = this.plotlyLegendOrColorBarWidth() + LEGEND_POINTS_MARGIN_RIGHT
+      const is_legend_elements_to_right_of_plotly_legend = this.isLegendElementsToRightOfPlotlyLegend(plotlyChart._fullLayout, config)
```

Comment:
Maybe you need to add a comment somewhere explaining that legend_elements include the legend points and bubble legend but not the ones that plotly draws (group and colorbar)

Reply:
Added description to `isLegendElementsToRightOfPlotlyLegend`